A new error:

```
FAILED | MEM 2.8/15.6 GB | DISK 52.8/71.6 GB tests/evaluate/test_default_evaluator.py::test_multi_classifier_evaluation[None-True] - AssertionError: assert {'confusion_m...lot.png', ...} == {'confusion_m...lot.png', ...}
  
  Extra items in the right set:
  'explainer'
  
  Full diff:
    {
        'confusion_matrix.png',
  -     'explainer',
        'per_class_metrics.csv',
        'precision_recall_curve_plot.png',
        'roc_curve_plot.png',
        'shap_beeswarm_plot.png',
        'shap_feature_importance_plot.png',
        'shap_summary_plot.png',
    }
```

--- a/mlflow/models/evaluation/evaluators/shap.py
+++ b/mlflow/models/evaluation/evaluators/shap.py
@@ -219,7 +219,7 @@ class ShapEvaluator(BuiltInEvaluator):
             _logger.debug("", exc_info=True)
             return
         try:
-            mlflow.shap.log_explainer(explainer, artifact_path="explainer")
+            mlflow.shap.log_explainer(explainer, name="explainer")
         except Exception as e:
             # TODO: The explainer saver is buggy, if `get_underlying_model_flavor` return "unknown",
             #   then fallback to shap explainer saver, and shap explainer will call `model.save`

--- a/mlflow/shap/__init__.py
+++ b/mlflow/shap/__init__.py
@@ -350,6 +350,7 @@ def log_explainer(
     """
 
     Model.log(
+        artifact_path=None,
         name=name,
         flavor=mlflow.shap,
         explainer=explainer,

--- a/tests/evaluate/test_default_evaluator.py
+++ b/tests/evaluate/test_default_evaluator.py
@@ -156,7 +156,7 @@ def test_regressor_evaluation(
     )
 
     assert json.loads(tags["mlflow.datasets"]) == [
-        {**diabetes_dataset._metadata, "model": model.metadata.model_uuid}
+        {**diabetes_dataset._metadata, "name": "dataset", "model": model.metadata.model_uuid}
     ]
 
     for metric_key, expected_metric_val in expected_metrics.items():
@@ -166,10 +166,6 @@ def test_regressor_evaluation(
             rtol=1e-3,
         )
         assert np.isclose(expected_metric_val, result.metrics[metric_key], rtol=1e-3)
-
-    assert json.loads(tags["mlflow.datasets"]) == [
-        {**diabetes_dataset._metadata, "model": model.metadata.model_uuid}
-    ]
 
     assert set(artifacts) == {
         "explainer",
@@ -270,7 +266,7 @@ def test_multi_classifier_evaluation(
         assert np.isclose(expected_metric_val, result.metrics[metric_key], rtol=1e-3)
 
     assert json.loads(tags["mlflow.datasets"]) == [
-        {**iris_dataset._metadata, "model": model.metadata.model_uuid}
+        {**iris_dataset._metadata, "name": "dataset", "model": model.metadata.model_uuid}
     ]
 
     assert set(artifacts) == {
@@ -365,7 +361,7 @@ def test_bin_classifier_evaluation(
         assert np.isclose(expected_metric_val, result.metrics[metric_key], rtol=1e-3)
 
     assert json.loads(tags["mlflow.datasets"]) == [
-        {**breast_cancer_dataset._metadata, "model": model.metadata.model_uuid}
+        {**breast_cancer_dataset._metadata, "name": "dataset", "model": model.metadata.model_uuid}
     ]
 
     assert set(artifacts) == {
@@ -456,7 +452,7 @@ def test_spark_regressor_model_evaluation(
     model = mlflow.pyfunc.load_model(spark_linear_regressor_model_uri)
 
     assert json.loads(tags["mlflow.datasets"]) == [
-        {**diabetes_spark_dataset._metadata, "model": model.metadata.model_uuid}
+        {**diabetes_spark_dataset._metadata, "name": "dataset", "model": model.metadata.model_uuid}
     ]
 
     assert set(artifacts) == set()
@@ -543,7 +539,7 @@ def test_svm_classifier_evaluation(svm_model_uri, breast_cancer_dataset):
         assert np.isclose(expected_metric_val, result.metrics[metric_key], rtol=1e-3)
 
     assert json.loads(tags["mlflow.datasets"]) == [
-        {**breast_cancer_dataset._metadata, "model": model.metadata.model_uuid}
+        {**breast_cancer_dataset._metadata, "name": "dataset", "model": model.metadata.model_uuid}
     ]
 
     assert set(artifacts) == {

--- a/tests/evaluate/test_default_evaluator.py
+++ b/tests/evaluate/test_default_evaluator.py
@@ -168,7 +168,6 @@ def test_regressor_evaluation(
         assert np.isclose(expected_metric_val, result.metrics[metric_key], rtol=1e-3)
 
     assert set(artifacts) == {
-        "explainer",
         "shap_beeswarm_plot.png",
         "shap_feature_importance_plot.png",
         "shap_summary_plot.png",
@@ -275,7 +274,6 @@ def test_multi_classifier_evaluation(
         "roc_curve_plot.png",
         "precision_recall_curve_plot.png",
         "shap_feature_importance_plot.png",
-        "explainer",
         "confusion_matrix.png",
         "shap_summary_plot.png",
     }
@@ -629,16 +627,19 @@ def test_pipeline_model_kernel_explainer_on_categorical_features(pipeline_model_
             evaluators="default",
             evaluator_config={"explainability_algorithm": "kernel"},
         )
-    run_data = get_run_data(run.info.run_id)
+    run_id = run.info.run_id
+    run_data = get_run_data(run_id)
     assert {
         # TODO: Uncomment once https://github.com/shap/shap/issues/3901 is fixed
         # "shap_beeswarm_plot.png",
         "shap_feature_importance_plot.png",
         "shap_summary_plot.png",
-        "explainer",
     }.issubset(run_data.artifacts)
 
-    explainer = mlflow.shap.load_explainer(f"runs:/{run.info.run_id}/explainer")
+    model = mlflow.search_logged_models(
+        filter_string=f"source_run_id='{run_id}' and name='explainer'", output_format="list"
+    )[0]
+    explainer = mlflow.shap.load_explainer(model.model_id)
     assert isinstance(explainer, _PatchedKernelExplainer)
 
 
@@ -4166,24 +4167,24 @@ def test_xgboost_model_evaluate_work_with_shap_explainer():
     xgb_model = xgboost.XGBClassifier()
     with mlflow.start_run():
         xgb_model.fit(X_train, y_train)
+        model_info = mlflow.xgboost.log_model(xgb_model, name="xgboost", input_example=X_train)
 
-        eval_data = X_test
-        eval_data["label"] = y_test
+    eval_data = X_test
+    eval_data["label"] = y_test
 
-        model_uri = mlflow.get_artifact_uri("model")
-        with mock.patch("mlflow.models.evaluation.evaluators.shap._logger.warning") as mock_warning:
-            mlflow.evaluate(
-                model_uri,
-                eval_data,
-                targets="label",
-                model_type="classifier",
-                evaluators=["default"],
-            )
-            assert not any(
-                "Shap evaluation failed." in call_arg[0]
-                for call_arg in mock_warning.call_args or []
-                if isinstance(call_arg, tuple)
-            )
+    with mock.patch("mlflow.models.evaluation.evaluators.shap._logger.warning") as mock_warning:
+        mlflow.evaluate(
+            model_info.model_uri,
+            eval_data,
+            targets="label",
+            model_type="classifier",
+            evaluators=["default"],
+        )
+        assert not any(
+            "Shap evaluation failed." in call_arg[0]
+            for call_arg in mock_warning.call_args or []
+            if isinstance(call_arg, tuple)
+        )
 
 
 @pytest.mark.parametrize(

--- a/tests/evaluate/test_default_evaluator.py
+++ b/tests/evaluate/test_default_evaluator.py
@@ -4165,16 +4165,19 @@ def test_xgboost_model_evaluate_work_with_shap_explainer():
     X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)
 
     xgb_model = xgboost.XGBClassifier()
-    with mlflow.start_run():
+    with mlflow.start_run() as run:
         xgb_model.fit(X_train, y_train)
-        model_info = mlflow.xgboost.log_model(xgb_model, name="xgboost", input_example=X_train)
 
+    logged_models = mlflow.search_logged_models(
+        filter_string=f"source_run_id='{run.info.run_id}'", output_format="list"
+    )
+    model_uri = logged_models[0].model_uri
     eval_data = X_test
     eval_data["label"] = y_test
 
     with mock.patch("mlflow.models.evaluation.evaluators.shap._logger.warning") as mock_warning:
         mlflow.evaluate(
-            model_info.model_uri,
+            model_uri,
             eval_data,
             targets="label",
             model_type="classifier",

